src/include/migraphx/generic_eval.hpp

```diff
@@ -0,0 +1,123 @@
+#ifndef MIGRAPHX_GUARD_MIGRAPHLIB_GENERIC_EVAL_HPP
+#define MIGRAPHX_GUARD_MIGRAPHLIB_GENERIC_EVAL_HPP
+
+#include <migraphx/module.hpp>
+#include <migraphx/stringutils.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/target.hpp>
+#include <migraphx/env.hpp>
+#include <migraphx/ranges.hpp>
+#include <migraphx/time.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/register_target.hpp>
+#include <iostream>
+#include <sstream>
+#include <algorithm>
+#include <set>
+#include <utility>
+#include <unordered_set>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+template <class F>
+void print_module(const module& m, F print_func)
```

Comment:
This should be a method of the module. It should be changed to `void print_module(std::function<void(instruction_ref, std::unordered_map<instruction_ref, std::string>)> print_func)`, rather than using a template.

Reply:
fixed